torch/testing/_internal/common_distributed.py

```diff
@@ -842,6 +852,66 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for DDP based Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
+        device_name = device.split(":")[0]
+        if hasattr(torch, device_name):
```

Comment:
I was thinking i've seen this code snippet before.  i found at least one case in profiler-

```
        if self.use_device and hasattr(torch, self.use_device):
            device_module = getattr(torch, self.use_device)
```

I think it'd be better to provide a utility for this perhaps in torch.device module.  Something like this

`def get_module(device: Union[str, torch.device]): -> Module`

Maybe something you could add in a follow up PR?

Reply:
@wconstab I think it already exists in utils https://github.com/pytorch/pytorch/blob/e6cddc9271fc7f3653ee0b2bedc7bd79fea41f5a/torch/_utils.py#L911

Reply:
@wconstab  : there are multiple places where the handle is retrieved from device , 
eg: torch/distributed/fsdp/_common_utils.py  
 _get_device_handle

Reply:
Agree that `_get_device_module` seems to be a more general way than calling:
`if TEST_ABC and torch.abc.device_count() > 0` everywhere.